sdk/communication/Azure.Communication.Identity/tests/CommunicationIdentityClient/CommunicationIdentityClientLiveTests.cs

```diff
@@ -106,6 +122,127 @@ public async Task GetTokenWithNullScopesShouldThrow()
+            }
+            catch (ArgumentException ex)
+            {
+                Assert.NotNull(ex.Message);
+                Assert.True(ex.Message.Contains(TOKEN_EXPIRATION_OVERFLOW_MESSAGE));
+                Console.WriteLine(ex.Message);
+                return;
+            }
+            Assert.Fail($"{nameof(CreateUserAndTokenWithOverflownExpirationShouldThrow)} should have thrown an exception.");
+        }
+
+        [Test]
+        [TestCase(MIN_VALID_EXPIRATION_TIME, TestName = "GetTokenWithMinValidCustomExpiration")]
+        [TestCase(MAX_VALID_EXPIRATION_TIME, TestName = "GetTokenWithMaxValidCustomExpiration")]
+        public async Task GetTokenWithValidCustomExpiration(string expiresAfter)
+        {
+            TimeSpan tokenExpiresAfter = TokenCustomExpirationTimes[expiresAfter];
+
+            CommunicationIdentityClient client = CreateClientWithConnectionString();
+            CommunicationUserIdentifier userIdentifier = await client.CreateUserAsync();
+            Response<AccessToken> accessToken = await client.GetTokenAsync(communicationUser: userIdentifier, scopes: new[] { CommunicationTokenScope.Chat }, tokenExpiresAfter: tokenExpiresAfter);
+
+            Assert.IsNotNull(accessToken.Value);
+            Assert.IsFalse(string.IsNullOrWhiteSpace(accessToken.Value.Token));
+
+            if (Mode == RecordedTestMode.Live)
+            {
+                var tokenExpirationValid = IsTokenExpirationValid(tokenExpiresAfter, accessToken.Value.ExpiresOn);
+                Assert.True(tokenExpirationValid);
+            }
+        }
+
+        [Test]
+        [TestCase(MIN_INVALID_EXPIRATION_TIME, TestName = "GetTokenWithMinInvalidCustomExpiration")]
+        [TestCase(MAX_INVALID_EXPIRATION_TIME, TestName = "GetTokenWithMaxInvalidCustomExpiration")]
+        public async Task GetTokenWithInvalidCustomExpirationShouldThrow(string expiresAfter)
+        {
+            try
+            {
+                TimeSpan tokenExpiresAfter = TokenCustomExpirationTimes[expiresAfter];
+                CommunicationIdentityClient client = CreateClientWithConnectionString();
+                CommunicationUserIdentifier userIdentifier = await client.CreateUserAsync();
+                Response<AccessToken> accessToken = await client.GetTokenAsync(communicationUser: userIdentifier, scopes: new[] { CommunicationTokenScope.Chat }, tokenExpiresAfter: tokenExpiresAfter);
+            }
+            catch (RequestFailedException ex)
+            {
+                Assert.NotNull(ex.Message);
+                Console.WriteLine(ex.Message);
+                return;
+            }
+            Assert.Fail($"{nameof(GetTokenWithInvalidCustomExpirationShouldThrow)} should have thrown an exception.");
+        }
+
+        [Test]
+        public async Task GetTokenWithOverflownExpirationShouldThrow()
+        {
+            try
+            {
+                TimeSpan tokenExpiresAfter = new TimeSpan(int.MaxValue / 20, 0, 0);
```

Comment:
For the future generations reading the code and asking themselves "what's so magical about 20?", I'd add a comment that the `int.MaxValue / 20` value can be used to initialize the `TimeSpan` but it makes it overflow....otherwise,

Reply:
Done.